test/quantization/test_qat.py

```diff
@@ -1912,6 +1898,78 @@ def test_qat_api_deprecation(self):
+            Float8FakeQuantizeConfig(torch.int8)
+        with self.assertRaisesRegex(
+            ValueError, "Please specify the granularity object instead of the class"
+        ):
+            Float8FakeQuantizeConfig(granularity=PerRow)
+        with self.assertRaisesRegex(
+            ValueError, "Expected PerRow or PerTensor granularity"
+        ):
+            Float8FakeQuantizeConfig(granularity=PerToken())
+
+    @parameterized.expand([(PerRow(),), (PerTensor(),)])
+    def test_float8_fake_quantize(self, granularity: Granularity):
+        """
+        Test that `Float8FakeQuantizer` is numerically close to `Float8Tensor`.
+        """
+        dtype = torch.float8_e4m3fn
+        fq_config = Float8FakeQuantizeConfig(dtype, granularity)
+        fake_quantizer = Float8FakeQuantizer(fq_config)
+        torch.manual_seed(self.SEED)
+        x = torch.randn(32, 64)
+        out = fake_quantizer(x)
+        out_expected = Float8Tensor.to_float8(x, dtype, granularity).dequantize()
+        sqnr = compute_error(out, out_expected)
+        self.assertGreater(sqnr, 16)
+
+    @parameterized.expand([(PerRow(),), (PerTensor(),)])
+    @unittest.skipIf(not _CUDA_IS_AVAILABLE, "skipping when cuda is not available")
+    @unittest.skipIf(not is_sm_at_least_89(), "Need sm89+")
+    def test_quantize_api_fp8_fp8(self, granularity: Granularity):
+        """
+        Test the following:
+            quantize_(model, QATConfig(Float8DynamicActivationFloat8Weight(), step="prepare"))
+            quantize_(model, QATConfig(Float8DynamicActivationFloat8Weight(), step="convert"))
+        """
+        torch.manual_seed(self.SEED)
+        m = M().to(torch.bfloat16).cuda()
+        example_inputs = (m.example_inputs()[0].to(torch.bfloat16).cuda(),)
+        base_config = Float8DynamicActivationFloat8WeightConfig(granularity=granularity)
+        quantize_(m, QATConfig(base_config, step="prepare"))
+        quantize_(m, QATConfig(base_config, step="convert"))
+        m(*example_inputs)
+
+    @unittest.skipIf(not _CUDA_IS_AVAILABLE, "skipping when cuda is not available")
+    @unittest.skipIf(not is_sm_at_least_89(), "Need sm89+")
+    def test_quantize_api_fp8_int4(self):
+        """
+        Test the following:
+            quantize_(model, QATConfig(Float8ActivationInt4WeightConfig(), step="prepare"))
+            quantize_(model, QATConfig(Float8ActivationInt4WeightConfig(), step="convert"))
+        """
+        torch.manual_seed(self.SEED)
+        m = M().to(torch.bfloat16).cuda()
+        example_inputs = (m.example_inputs()[0].to(torch.bfloat16).cuda(),)
+        base_config = Float8ActivationInt4WeightConfig(group_size=128)
+        try:
+            quantize_(m, QATConfig(base_config, step="prepare"))
+            quantize_(m, QATConfig(base_config, step="convert"))
+            m(*example_inputs)
+        except ImportError as e:
+            if "fbgemm-gpu-genai" in str(e):
+                self.skipTest("fbgemm-gpu-genai not available")
```

Comment:
nit: we can skip the test when fbgemm-gpu-genai is not installed: https://github.com/pytorch/ao/blob/1dca63818d7e7cd96b403657aa0dde6170a560da/test/quantization/quantize_/workflows/int4/test_int4_preshuffled_tensor.py#L45